### DISCUSSION TITLE
Some ideas for new updates

### DISCUSSION THREAD

**Discussion by javegon** [bangcorrupt/hacktribe#209] [Ideas]
UPDATED (28 Feb 2023)

Hi everyone 
I was thinking about nice features can be update for next firmwares and maybe they are not very difficult to implement , I have no idea of coding ... maybe Mr. @bangcorrupt  can reply and say how difficult it is.

1- Filters: The are some filter, but all of them are 12db or 6 db per octave (I am not sure which one). Could be great to have different filters. 24db per octave or even 48db. Also add some different filters like Formant, Comb.... I don't really Know how difficult could be to improve that section. Also a better control of filters, the volume increase when it is used, that means if we want to activate and desactivate it the volume will be affected. Very Annoying.  Might it be able to recorded in motion section like Mfx send button and ifX On. (same for Amp EG button) ?

2- Shift:  Another idea is add new options / combination for shift button in order to have more possibilities.  Some Knobs doesn't have double / triple function, and It could get that options for more control. 2nd ifx or get control of specific  parameter for ifx.  Shift + Edit OSC to get control of the end point of the sampler. Shift + Attack to control a Release .... they are just an example.

3- Master FX: Some of those fxs are useless or misconfigured like Room Reverb, Hall Reverb, Jag Filter, I can't hear the difference when I am playing a track. Could be possible a makeup for some of them? Tape delay could be in ping pong like mod (stereo )

4 - IFX: Same, a way to update some fx to make it better. For example, Distortion some times is not enough because works an like analogue fx as more signal more distortion, low volume signal no distortion at all, but the volume increases a lot as well, if you use it in low volume is not affected by distortion at all. Could be nice to have a better balance, I guess. 
Copy Mod to ifx - add lfo options to some routing from Mod parameter in order to have more than one LFO Modulators. 
Add some from Master section, imposible Reverbs by the fact of the dependence of the processor, but what about Mod Delay, and would be nice also modify a bit delays in Ifx, they sound very flat, it would be great if delays have filter on it maybe low pass filtering helps to make delay more natural in the tails, while the delay is fading away the sound becomes less bright, like a band pass filter, also to get longer delays it would be great.
Effects combination - for example distortion + delay, delay + phaser ...

5 - ARP: To have a real arpeggiator it will be another cool tool.

6 - MIDI: I don't know if exist a chart with all midi parameters to control the Electribe.  I think Cutoff CC is the same Midi CC for all parts. It will be great to assign different Midi CC in order to control each part of 16 with a external Midi Controller. In order to being able to control 16 parts volume, 16 knobs filter per each part, panoramic, ifx, or even have macro controls in an external device. Even we could go farther if we assign parameter to touch pad for example x for filter y for pitch. 
Another idea is that if we have a full map of CC we are able to create a layout to control Electribe with all the parameters using  a software or  control them with a layout in Chrome using midi data. 

7 - Modulation: 2nd Modulation with shift button. And also more mod.types like send de modulation to Decay/Release. Other nice feature will be send gate arp to Filter or EG Int or example as Modulator.  

8 - Steps Sequencer: 8bars/16 bars the most dreamed feature for this machine. With shift button we could add a functionality to sequencer button or maybe keep pushed + num pad, and have 16 bars, on bar in each num Pad. And activated how many bars we want or moving through them.

**NOT POSSIBLE** _9 - Memory: 26mb sometimes are not enough , There is any way to update it ? by software or hardware_ 

10 - Stereo: Use just one part for stereo sounds

**NOT POSSIBLE** _11 - More Outputs: I saw recently the update of MPC One, now it is possible to use a sound card usb connected and use the outputs of the sound card._ 

12 - Drumkits : Will be possible to a have the options to work samples assing note, I mean, when we use slices option, the slice take each slice for each step, but could be great if we could use per note as well, in this way we will able to work with drum kits. Or maybe works in other better way, assigning a sample per note or slice per note or ...
The trick of set different start points doesn't work smooth and fine enough in order to work with multi-sampling as a drum kit.

13 - Choke Groups: Now we have choke in 13-14 and 15-16. Could it be extended for more parts?

14 - Sync Unit: Now we have 1 and 2 step, More is possible? like 1/2 half and 3 step and 4 step

15 - Menu: Have Bank Presets for Sounds

16 - Solo Button: Have a momentary solo button like Electribe MX/SX. When we push twice Part Mute change to solo, but if we wan to back, we have to remember what part were active and push all together, it is a bit annoying. Have a momentary (when is pressed) solo button could be another nice feature.

17 - Change Key note: As Electribe MX/SX it was possible to change the root of the melody. In Electribe 2 we can do It through the menu and Knob changing the key note, but it is really difficult to change from C to F# when we are playing in live, Electribe MX/SX had a nice feature using the 12 pads (7 notes + 5 sharps), each one for one note and his #. C and his C# 

_/////// NEW /////// Confirmation answer when we want to change the key note could be a good and easy solution- If we want to make changes in the fly is easier first select the key and after confirm than the changes affect notes in between._

18 - Gate Arp - As Modulator for ifx, filter, etc. Like step modulator 

19 - EQ - in ifx Section to have another nice and very useful could be eq cut frees

**20 - LFO -** 
20.1 - FM Synthesis - the theory about FM is that one OSC is affected by another one, normally Sine waves, so we can use the LFO Modulation as modulator and OSC as carrier. In theory we could be able to do it, or a kind of. Only it should add new Sine LFO, and maybe also enlarge the range of LFO speed, it could be possible affecting the Pitch or the Phase (better choice) of the carrier.  [ link about FM synthesis](https://support.apple.com/en-gb/guide/logicpro/lgsife418213/mac ) 
20.2 - Create a LFO Sine. 
20.3 - New Modulation to Decay/Release.
20.4 - Second Modulation with " shift "
20.5 - Pattern Modulation. Add new pattern modulation to scape from LFOs. Could be great modulate filter, pitch or other stuff with a pattern like scales, rhythms ...
20.6 - I saw another devices that have very interesting LFO shapes, Chromatic, Major Scales, Minor Scale, exponential, quad ramp, 3 step, 4 steps, 8 steps, 16 steps, Buzz.... Those lfo can be use as filter or even as arpeggiator setting with pitch and 3 step (1st, 3rd, 5th) as chord notes.

21 - IFX - Add new combined effects, the algorithm are already made. So, I am wondering how difficult could be add new delay + Decimator or another ones like Delay + Filter. Or Maybe Sine LFO Pitch,  Sine LFO Filter, S&H LFO Filter, Stairs LFO Scale C Pitch (Where the edit also can be controlled by Mod LFO)

22 - Groove types - Add some grooves from AKAI - There are some groove very useful for electronic music that it might be added. Also it would be very interesting been able to record and change groove types per bar or maybe create new ones for 4 bars. Or even assign it for filter, decay, pitch or ifx.

23 - Master FX - Could be possible to create a short cut - Master Fx + Pads - To create a fx layer ? Assigning a specific fx..  loop,  filter, stop vinyl ... etc in each of the pads in order to play them in live

24 - Part Chain - Could be possible chain parts in order to have 3 osc synth ? Maybe could be a nice feature. Now it posible coping the part to other ones, but it is not possible to play on a fly.

25 - Pattern Knob - When we want to move us to other pattern always it changes at the end of the sequence, even if we didn't find the pattern that we wanted, Could we have a change for that ? Change to the next partern at the end of the sequence only when we leave to touch or move the knob pattern. Some times I want to move from pattern 10 to pattern 60, I should move the knob really really fast .... and I never get it. (that is just an example) 
this would be a good fit.

26 - More options per button - Filter button has 3 kind of filters, I call it carousel. Could we have more carousel option for other parameters? For example Gate Arp - gate arp affect to volume, if we press twice gate arp would affect filter, three times pitch ... and so on. 7 times will be off. Also moving us with the options we find different kind of gate, we could add off , and have available 6 types or gate arps (affecting volume, filter, pitch) and in the end of each one am off option.

27 - Blink - Part Mute has to options. One normal one blink. We could add more options to some button adding a blink.

28 - Event Recording - Some of the options like cutoff, trigger, touch pad, keyboard are available while we reproduce the file, but mute and unmuted is not allowed. Could  that option be changed ? 

_//////// NEW ///////  - 29 - Synth Chain Part - Chain part to get a 2 or more osc synth in one part._

THANKS !!!!

**Comment by bangcorrupt** [comment]
Hi.  There are some good ideas here, some of them are planned, some of those may actually happen someday.  FX editing and shortcuts are in progress, drum-kits and choke groups have been discussed.  

Increasing sample memory is very unlikely, most of the DSP RAM is used for sample data already.  It may be possible to have 'quick load' samples stored in the CPU RAM, but this won't increase the actual sampling time.  

Attaching an audio interface won't happen in Hacktribe, electribe 2 uses a RTOS, not a Linux kernel.   It may be possible to attach a multichannel DAC and switch the DSP output to TDM, but I don't know if the board would cope with the increased bitrate.

**Reply by javegon** [reply]
Thanks for the feedback, I will update this list in order to make a full sheet of ideas. 
I hope some of them could be possible next updates of hacktribe.

About that "It may be possible to attach a multichannel DAC and switch the DSP output to TDM, but I don't know if the board would cope with the increased bitrate" Could be possible if the bitrate is not too high ? Wooow it would be so great to have different outputs for sounds. 

What about other improves, what are your thoughts about ... add Sine LFO and increase the range of speed, drums kit (different sounds for different notes) to be played in only one part or assigning Grooves / Gates to affect filter, decay , distortion, edit button...

Regards

**Reply by bangcorrupt** [reply]
> Sine LFO

There is a sine LFO hidden in the factory firmware, it will be available in Hacktribe when I've fixed some bugs.

**Reply by bangcorrupt** [reply]
> IFX

Balancing the level of the distortion will be possible by editing the control mapping in the FX preset.  There are also EQ bands built into the distortion module allowing for some fairly extreme tone shaping.  The insert delays have high and low pass filters in the feedback paths, we will be able to set these in the preset and map them to the knob if desired. Most of the IFX modules occupy a double slot, only a few of them (EQ, overdrive, filter, cheap comp, punch) can be combined.

**Reply by javegon** [reply]
thanks for the answer @bangcorrupt
If any ideas come to my mind I will write down here in order to give it fresh new features, always trying to give priorities to develop simple updates, or combine functions that korg already has. ;)
Best wishes for this new year 2023

**Comment by Forumlivii** [comment]
Ciao, a proposito di filtri. Se possibile inserire i filtri dei mixer da dj. Filtri Xone Allen & Heath, e filtri Pioneer

**Reply by bangcorrupt** [reply]
> Hello, speaking of filters. If possible insert the filters of the dj mixers. Allen & Heath Xone filters, and Pioneer filters

Hi.  As far as I know, those mixers have analog filters, so it is not possible.

**Reply by javegon** [reply]
Could 24db filters be added in Electribe?

**Reply by bangcorrupt** [reply]
The Moog, FX, and maybe e2 low pass filters are 24dB/Oct.  I think there are a couple of hidden filters with gentler roll off I'll try to enable at some point.

**Reply by javegon** [reply]
I didn't know e2 low pass filter is 24db/oct. 
I thought it was 6db/oct or 12db/oct. When I listen it on E2 and after in my computer on a virtual synth they sound very different.
Thanks

**Comment by Forumlivii** [comment]
Thanks, it's very kind of you

**Reply by bangcorrupt** [reply]
Thank you for sponsoring Hacktribe!   :smiley:

**Comment by Forumlivii** [comment]
thanks to you ☺️ expected date for the next firmware ?

**Reply by bangcorrupt** [reply]
I don't know when, see #157 for a vague plan.

**Comment by Forumlivii** [comment]
ok good job 💪🏻

**Comment by awchx** [comment]
I have some ideas how to manage more than 4 bars per pattern (idea no. 8), if that feature ever turns out possible.

8 bars: actually not my idea, but I find it very good. Shift+BarNumber would select bars 5-8, indicated with blinking LEDs as opposed to solid ones for 1-4. Straightforward and elegant.

16 bars: my idea is to have 4 banks of 4 bars. Shift+BarNumber would select a bank, while pressing just the number would select a bar within that bank.
[1 2 3 4] [5 6 7 8] [9 10 11 12] [13 14 15 16]
The tricky part is how would it be indicated, so I came up with LED blinking patterns, assisted with an info on the LCD display.
The blinks would be as follows: one short blink (1/4 of the time) indicating the bank number, followed by a long blink (the remaining 3/4) showing the position inside that bank.
For example: one short blink of the button 3 would indicate that the third bank is being used [9 10 11 12], while the following long blink of the button 2 would show us that we're currently on the bar number 10. [9 (10) 11 12].

64 bars: if it ever turns out to be possible to have patterns that long, I'd make use of the fact that the e2 has a set of 16 pads, neatly organized in groups of 4. In this case, holding the bar number button would let us choose the bank 1-4, while hitting a pad would switch to a bar within that bank.
Indicating it would be a real challenge, since we have only 4 LEDs available, so I'd suggest to rely on the LCD display. The LEDs would still show us a general indication of which bank is active, for example all the LEDs turned on at once would indicate that the 64 bar mode is active, while the corresponding LED would blink, showing us the number of the current bank.

Huge thanks to Bangcorrupt and the Hacktribe/Freetribe team for keeping the e2(s) alive, you rule!

**Comment by awchx** [comment]
Regarding idea no. 2: the Shift+OscEdit combo could be used to tidy up FM and VPM oscillators by grouping them together as more general types: X-SAW, X-SQUARE, X-TRI, X-SINE, VPM-SAW, VPM-SQUARE, VPM-TRI, VPM-SINE (8 in total) and moving the ratio to Shift+OscEdit.
